src/makeCalculator.js

```diff
@@ -4,7 +4,45 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add: function (valueResult) {
+      return function (value) {
+        return valueResult + value;
+      };
+    },
+
+    subtract: function (valueResult) {
+      return function (value) {
+        return valueResult - value;
+      };
+    },
+
+    multiply: function (valueResult) {
+      return function (value) {
+        return valueResult * value;
+      };
+    },
+
+    divide: function (valueResult) {
+      return function (value) {
+        return valueResult / value;
+      };
+    },
+
+    operate: function (callback, value) {
+      this.result = callback(this.result)(value);
+
+      return this;
+    },
```

Comment:
[CODE KNOWLEDGE] - Use the method shorthand syntax for the `operate` method as well.